samples/nrf9160/gnss/src/assistance_supl.c

```diff
@@ -31,6 +31,9 @@ static ssize_t supl_read(void *p_buff, size_t nbytes, void *user_data)
 
 	if (rc < 0 && (errno == EAGAIN)) {
 		return 0;
+	} else if (rc == 0) {
+		/* Peer closed the socket, return an error. */
+		rc = -1;
```

Comment:
I guess this would be more aligned in terms of style:
```suggestion
	if (rc < 0 && (errno == EAGAIN)) {
		rc = 0;
	} else if (rc == 0) {
		/* Peer closed the socket, return an error. */
		rc = -1;
```
And check same spot in `gnss_supl_support.c` also.